### DISCUSSION TITLE
Small reorganization of cards at storage section

### DISCUSSION THREAD

**Discussion by ancorgs** [agama-project/agama#1327] [Ideas]
During a conversation with @dgdavid we came to the realization that it would be good to substitute the current cards "Find Space" and "Result" by "Actions" and "Final Layout". They say an image is worth a thousand words, so...

![success](https://github.com/openSUSE/agama/assets/3638289/513e265e-724f-4741-8559-30050df2b2d9)

The "Change" button at the Actions card would lead to the screen to change the space policy.

The structure of three items at the Actions card would be kept even when the proposal fails.

![failed](https://github.com/openSUSE/agama/assets/3638289/7add5186-8a46-4c1a-ab5f-9934a3d0a41d)

There are many visual decisions to be taken (colors, wording, usage of `<strong>`...), but the essence of the first version is already visible in the screenshots.

The wording (or at least the overall message) of the three lines is well defined for most situations.

- First line if there is no proposal or if there is a proposal but no delete actions are planned (one of these):
  - Destructive actions are allowed
  - ~~Some destructive actions are allowed (_likely this is going too fine-grained and the previous message is enough_)~~
  - Destructive actions are not allowed
- First line if there is proposal and some delete actions are planned (one of these):
  - There are 2 destructive actions planned
  - There are 2 destructive actions planned affecting Windows, openSUSE Tumbleweed
- Second line if there is no proposal (one of these)
  - Shrinking some partitions is allowed
  - Shrinking partitions is not allowed
- Second line if there is a proposal but no resize actions are planned (one of these):
  - Shrinking some partitions is allowed but not needed
  - Shrinking partitions is not allowed
- Second line if there is a proposal and some resize actions are planned (one of these):
  - 3 partitions will be shrunk
  - 3 partitions will be shrunk affecting Windows, openSUSE Tumbleweed
- Third line if there is no proposal
  - Cannot accommodate the required file systems for installation
  - Check the 10 planned actions

**Comment by ancorgs** [comment]
The screenshots at the description are taken from this branch, that can hardly be used as a base for something since it's extremely hacky (only intended for prototyping). https://github.com/openSUSE/agama/compare/new-ui-proposal...ancorgs:agama:show-idea-actions?expand=1

**Comment by ancorgs** [comment]
Implementation note. This needs some extension in the API because there are fields to indicate the removed systems and to mark the destructive actions, but there are no equivalent ones for the resized systems and the resize actions.

**Comment by dgdavid** [comment]
This has been addressed at https://github.com/openSUSE/agama/pull/1354. Closing as resolved.
